logger/logger.go

```diff
@@ -13,11 +13,14 @@ type causer interface {
 
 var GlobalFields = logrus.Fields{}
 
-type Logger func(Valuer, error) *logrus.Entry
+type Logger func(Valuer, ...error) *logrus.Entry
 
 func New(name string) Logger {
-	return func(ctx Valuer, err error) *logrus.Entry {
-		return ContextLog(ctx, err, nil).WithField("component", name)
+	return func(ctx Valuer, err ...error) *logrus.Entry {
+		if len(err) == 0 {
```

Comment:
How about a `switch len(err)` and barf an error if it's not 0 or 1?

Reply:
barf as in panic?

Reply:
Perhaps not a panic, but at least log an Error, since you are already within a logger.

Reply:
Updated, please have a look.

Reply:
I'd be more inclined to expand the variadicity to `ContextLog`, and add a field for each error.  Actually respect the signature.

Reply:
What would we name those errors? `error1`, `error1` etc (where the number is the same as the index number, with 0 being omitted)?

Reply:
Yep, I like that (assuming you mean `error, `error1`).  Sorry I missed this comment earlier, github didn't load it for me.